### PR TITLE
feature/OE-210-region-specific-folder-management

### PR DESCRIPTION
**Pull request by devsquad-mathews-lima**
## Review

For the submitter, initial self-review:

- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency between the changeset and the goal stated above
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Complexity - is the proposed change introducing any unjustified complexity? 
- [ ] Thoroughness of added tests and any missing edge cases

### CODE REVIEW

File: app/Models/TrainingPageSection.php [hunk 1]
```diff
@@ -64,4 +81,9 @@ public function scopeSearch(Builder $query, $search)
             );
         });
     }
+
+    public function isDepartmentSection()
+    {
+        return $this->department_folder === true;
+    }
```
Comment (Ron-Junior) [inline app/Models/TrainingPageSection.php:88]: department_folder property needs to be compared with true? wouldn't that be boolean?

File: app/Http/Livewire/Castle/ManageTrainings/Trainings.php [hunk 2]
```diff
@@ -112,25 +112,20 @@ public function changeDepartment()
...
+            ->when($this->search === '', function ($query) use ($section) {
+                $query->where('training_page_sections.parent_id', $section->id ?? 1);
+            })
+            ->when($this->search !== '', function ($query) {
+                $query->where(function ($query) {
+                    $query
+                        ->orWhere('training_page_sections.title', 'like', "%{$this->search}%")
+                        ->orWhere('training_page_contents.description', 'like', "%{$this->search}%");
+                });
+            })
```
[the remark this answers is not in the document]
Comment (devsquad-mathews-lima) [reply app/Http/Livewire/Castle/ManageTrainings/Trainings.php:127]: yeah, that's a good idea.

File: app/Models/TrainingPageSection.php (hunk 1 above)
Comment (devsquad-mathews-lima) [reply app/Models/TrainingPageSection.php]: not necessarily, but sometimes I like to make things more explicit :detective:

**Review by Ron-Junior — APPROVED** [review]
